### ISSUE TITLE
Bug Report: Character Select animations abruptly restart when switching characters

### ISSUE THREAD

**Issue by Michu0204** [FunkinCrew/Funkin#3536]
### Issue Checklist

- [X] I have properly named my issue
- [x] I have checked the Issues/Discussions pages to see if my issue has already been reported

### Platform

Itch.io (Downloadable Build) - Windows

### Browser

None

### Version

0.5.1

### Description (include any images, videos, errors, or crash logs)

When you go to the playable characters select menu and switch the characters (for example from boyfriend to pico or pico to boyfriend, the ??? characters too), the animation of boyfriend and girlfriend (or nene and pico) restarts after 1 second, it happens only sometimes tho

https://github.com/user-attachments/assets/14d57b50-a9b0-4081-ad5a-887f50d34859

### Steps to Reproduce

1. Open the game
2. Open the freeplay menu
3. click tab (to open the playable characters select menu)
4. switch characters like 2 times

**Comment by EliteMasterEric** [comment]
This happens because they bop to the beat of the music, and you're switching off-beat.

Not sure what the best solution is here.

**Comment by SnowyyWHY** [comment]
> This happens because they bop to the beat of the music, and you're switching off-beat.
> 
> Not sure what the best solution is here.

maybe make it slow at the start and speed up so it bops to the beat

**Comment by Lasercar** [comment]
> This happens because they bop to the beat of the music, and you're switching off-beat.
> 
> Not sure what the best solution is here.

Not really, it's actually because there isn't a way to swap the current animation of an altas sprite without playing the animation, and it just so happens that the idle animation is also the one used for the beat.

There's a code comment about in BF and GF's character select class:

>    // TODO: There's a minor visual bug where there's a little stutter.
>    // This happens because the animation is getting restarted while it's already playing.
>    // I tried make this not interrupt an existing idle,
>    // but isAnimationFinished() and isLoopComplete() both don't work! What the hell?
>    // danceEvery isn't necessary if that gets fixed.

**Comment by Michu0204** [comment]
I think this could be fixed too by making the other anims run in background? so like when you return to boyfriend its on the exact frame of the anim to the music

**Comment by Starexify** [comment]
ts needs fixed fr, I think the character select state contains the animation play issue ?
